app/controllers/api/v1/merchants/revenue_controller.rb

```diff
@@ -1,11 +1,17 @@
 class Api::V1::Merchants::RevenueController < ApplicationController
   def index
-     result = ActiveRecord::Base.connection.execute("SELECT m.id, SUM(ii.quantity * ii.unit_price) AS revenue FROM merchants m JOIN invoices v ON m.id = v.merchant_id JOIN invoice_items ii ON ii.invoice_id = v.id JOIN transactions t ON t.invoice_id = v.id WHERE t.result = 'success' AND v.status = 'shipped' GROUP BY m.id ORDER BY revenue DESC LIMIT #{params[:quantity]};")
+    return nil if params[:quantity].to_i < 1
 
-     merchants = result.pluck("id").map do |id|
-       Merchant.find(id)
-     end
+    if request.env["PATH_INFO"].split('/').last == "most_items"
```

Comment:
Endpoints most_revenue and most_items hit same controller with same params[:quantity]. Utilized HTTP request path to differentiate which API endpoint user wants.